bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -152,6 +153,9 @@ void ReplicaImp::registerMsgHandlers() {
                                    bind(&ReplicaImp::messageHandler<ReplicasRestartReadyProofMsg>, this, _1),
                                    bind(&ReplicaImp::validatedMessageHandler<ReplicasRestartReadyProofMsg>, this, _1));
 
+  msgHandlers_->registerMsgHandler(MsgCode::StateTransfer,
```

Comment:
I have a small doubt: 
https://github.com/Liorgal28/concord-bft/blob/173a8f86900f67708e20b810cbaa1341c509ff49/bftengine/src/bftengine/ReplicaForStateTransfer.cpp#L46
if msgHandlers_ is common then there might be a race to set the handler.

Reply:
You're right.
Removed from here but I had to add msg handler to the RO replica too since it also derives from ReplicaForStateTransfer.